router/factory.go

```diff
@@ -31,14 +18,39 @@ type Factory struct {
 	ProcErrorDB      jobsdb.JobsDB
 	TransientSources transientsource.Service
 	RsourcesService  rsources.JobService
+	Logger           logger.Logger
+	Stats            stats.Stats
 }
 
 func (f *Factory) New(destination *backendconfig.DestinationT, identifier string) *HandleT {
+	throttlerFactory, err := throttler.New()
```

Comment:
Why would we need a different `throttler.Factory` for each router? Wouldn't it be simpler if we created only one `throttler.Factory` while creating the `router.Factory` in `embeddedAppHandler.go` & `embeddedAppHandler.go` and fail to `StartRudderCore` if an error occurs?